### PR TITLE
Html_css_popup_task

### PR DESCRIPTION
**Pull request by unabyband**
### Html_css_popup_task

[Demo](https://unabyband.github.io/HTML-CSS-Popup/) |
[Code base](https://github.com/unabyband/HTML-CSS-Popup)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/unabyband/popup-task/main.css [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    border-top: 2px solid #e5e5e5;
+    display: flex;
+}
+/*Приховання кнопки More при натисканні*/
+.hidden_input:checked ~ .morebutton {
+    display: none;
+}
+/*---------------------------------------------------------------------------
+Прив'язка зображень іконок до видимих елементів попапа*/
+.popup_window_visible .popup_window_list__item:first-child .popup_window__img {
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:217]: Semantically this is not the correct use of images, this is not a background in this case, but icons. So, you need to move it to an HTML file as `img` tags

File: submissions/unabyband/popup-task/main.css [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+  border-width: 0 8.5px 8.5px;
+  right: 102.5px;
+  border-bottom-color: rgba(0, 0, 0, 0.2);
+  top: -10px;
+}
+.popup:after {
+  border-color: transparent;
+  border-bottom: solid #fff;
+  border-style: dashed dashed solid;
+  border-width: 0 8.5px 8.5px;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:104]: 8.5 px? wow, why do you need such accuracy?

**Review by kasionio — CHANGES_REQUESTED** [review]
Nice job @unabyband ! All works as expected. Let's make your codebase and design even better ;)
First of all, no need to leave comments. It's a good practice to add comments for some parts that can be not obvious to you in a month or other developers :)

Icons are jumping in hover effect, you can add a transparent border by default to prevent it

File: submissions/unabyband/popup-task/main.css (hunk 2 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:102]: Just tried to align it in the center of the icon %)

File: submissions/unabyband/popup-task/main.css (hunk 1 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css]: Done! 
Plus made a  few Photoshop manipulations over icons files in order to make them square.

**Comment by unabyband** [comment]
Well, I've made all edits, deleted bunch of comments and hope it's ok now.
Thanks for review.

File: submissions/unabyband/popup-task/main.css [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    transition: .2s;
+    color: #80868b;
+    font-size: 14px;
+}
+
+.find_field-icon {
+    width: 20px;
+    height: 30px;
+    position: absolute;
+    top: 5px;    right: 15px;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:258]: try to use some prettifier

**Review by kasionio — CHANGES_REQUESTED** [review]
Good job! I see some issues that need to be fixed in design. look at screenshots, please.
1. Too big space between divider and top items
2.  Image (container) has no padding.
3. Link text has a redundant underline
4. Bottom link has a weird focus 
5. Fix mobile view
image
image
image

File: submissions/unabyband/popup-task/index.html [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                </div>
+                <div class="top_menu__item-container">
+                    <input class="input_popup" type="checkbox" id="moreoption" name="moreoption">
+                    <label class="button_popup" for="moreoption"></label>
+                    <div class="popup">
+                        <div class="popup_window">
+                        <ul class="popup_window_list popup_window_visible">
+                        <li class="popup_window_list__item">
+                            <a href="#" class="popup_window__active">
+                                <span class="popup_window__img"><img class="popup_window__img-src" src="./icons/calendar.png" alt=""></span>
```
[the remark this answers is not in the document]
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html]: Done!

File: submissions/unabyband/popup-task/index.html [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                        <li class="popup_window_list__item">
+                            <a href="#" class="popup_window__active">
+                                <span class="popup_window__img"><img class="popup_window__img-src" src="./icons/documents.png" alt=""></span>
+                                <span class="popup_window__label">Document</span>
+                            </a>
+                        </li>
+                        </ul>
+                        <input class="hidden_input" type="checkbox" id="showidden" name="moreoption">
+                        <label class="morebutton" for="showidden">More</label>
+                        <div class="divider"></div>
```
[the remark this answers is not in the document]
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html]: Done!

File: submissions/unabyband/popup-task/main.css (hunk 3 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css]: Ok, done!

**Comment by unabyband** [comment]
I've fixed all noticed issues and made some responsive features for my task. 
Please re-review, and hope everything is ok now.
Have a great Independence Day :)

File: submissions/unabyband/popup-task/index.html [hunk 6]
```diff
@@ -0,0 +1,197 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+        <link rel="stylesheet" href="css/main.css">
+        <title>unabyband pop-up</title>
+    </head>
+    <body>
+        <div class="header">
```
Comment (kasionio) [inline submissions/unabyband/popup-task/index.html:10]: Too a lot of divs :) Let's make the markup more semantic friendly with this:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header

File: submissions/unabyband/popup-task/index.html [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+            <div class="top_menu">
+                <div class="top_menu__item-container">
```
Comment (kasionio) [inline submissions/unabyband/popup-task/index.html:12]: list of items starts here, not only in popup

File: submissions/unabyband/popup-task/index.html [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+                </div>
+                <div class="top_menu__item-container">
+                    <input class="input_popup" type="checkbox" id="moreoption" name="moreoption">
+                    <label class="button_popup" for="moreoption"></label>
+                    <div class="popup">
+                        <div class="popup_window">
+                        <ul class="popup_window_list popup_window_visible">
+                        <li class="popup_window_list__item">
+                            <a href="#" class="popup_window__active">
+                                <div class="popup_window__img">
```
Comment (kasionio) [inline submissions/unabyband/popup-task/index.html:26]: div looks redundant here, you can handle it without wrapping the image

File: submissions/unabyband/popup-task/main.css [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+    padding-top: 0px;
+    margin-bottom: 0px;
+}
+
+.popup_window_hidden {
+    display: none;
+    margin-top: 0px;
+}
+
+.popup_window__active {
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:149]: why `active`?

File: submissions/unabyband/popup-task/main.css [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+.popup_window_list__item:hover {
+    border: 1px solid rgba(0, 0, 0, 0.75); 
+}
+
+.popup_window__img {
+    height: 60px;
+    width: 60px;
+    display: inline-block;
+}
+.popup_window__img-src {
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:172]: why img-src? src means source

File: submissions/unabyband/popup-task/index.html [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+            </div>
+        </div>
+        <div class="find">
+            <div class="find-container">
+                <div class="find_cover">
+                    <img class="find_cover-image" src="icons/cover.jpg" alt="Cover image">
+                </div>
+                <div class="find_field">
+                    <input type="text" placeholder="Find yourself or something else..." class="find_field-input">
+                    <span class="find_field-icon"></span>
```
Comment (kasionio) [inline submissions/unabyband/popup-task/index.html:192]: this an example of good naming 👍

File: submissions/unabyband/popup-task/main.css [hunk 12]
```diff
@@ -0,0 +1,302 @@
+html, body {
+    font-family: Arial, Helvetica, sans-serif;
+    width: 100%;
+    height: 100%;
+    margin: 0;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:5]: for resetting default styles, you can use 
*{margin: 0; padding: 0;}
 It is a common practice
https://www.webfx.com/blog/web-design/css-tip-1-resetting-your-styles-with-css-reset/

File: submissions/unabyband/popup-task/main.css [hunk 13]
```diff
@@ -0,0 +1,302 @@
...
+    background: url("../icons/bell.png") center center no-repeat;
+    background-size: 20px 25px;
+}
+
+.top_menu__item_user {
+    display: block;
+    height: 32px;
+    width: 32px;
+    border-radius: 50%;
+    background: url("../icons/favicon.png") center center no-repeat;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:50]: seems one `center` is redundant :)

File: submissions/unabyband/popup-task/main.css [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+    background-size: 32px 32px;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:51]: why do we need this?

File: submissions/unabyband/popup-task/main.css [hunk 15]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+.top_menu__item_bell:hover, .top_menu__item_user:hover, .button_popup:hover {
+    opacity: 75%;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:55]: no, bad styling idea, better remove it :) So, you are doing items worse visible to the user

File: submissions/unabyband/popup-task/main.css [hunk 16]
```diff
@@ -0,0 +1,302 @@
...
+    height: 430px;
+    position: absolute;
+    box-sizing: border-box;
+    top: 62px;
+    right: 15px;
+    border: 1px solid #ccc;
+    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.2);
+    background: #ffffff;
+    flex-direction: column;
+    z-index: 50;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:80]: I preffer 77 :D

File: submissions/unabyband/popup-task/main.css [hunk 17]
```diff
@@ -0,0 +1,302 @@
...
+    width: 320px;
+    height: 430px;
+    position: absolute;
+    box-sizing: border-box;
+    top: 62px;
+    right: 15px;
+    border: 1px solid #ccc;
+    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.2);
+    background: #ffffff;
+    flex-direction: column;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:79]: it is easier to read if for example flex properties will be placed one by one

File: submissions/unabyband/popup-task/main.css [hunk 18]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+.find_field-input:focus {
+    border: 2px solid rgba(0, 0, 0, 0.75);
+}
+.find_field-input:hover {
+    background-color: rgba(0, 0, 0, 0.1);
+}
+
+@media screen and (max-width: 680px) {
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:283]: better to use common devices breakpoints 
https://www.lambdatest.com/blog/how-to-use-css-breakpoints-for-responsive-design/

**Review by kasionio — CHANGES_REQUESTED** [review]
Great job! Much better now. But still need some improvements :)
Need to improve borders on focus state from a keyboard and mouse, take a look. And also padding of container

image

File: submissions/unabyband/popup-task/main.css (hunk 2 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:102]: Done

File: submissions/unabyband/popup-task/index.html (hunk 11 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html:192]: Thanks

File: submissions/unabyband/popup-task/main.css (hunk 15 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:55]: Done!

File: submissions/unabyband/popup-task/main.css (hunk 18 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:283]: Done

File: submissions/unabyband/popup-task/main.css (hunk 13 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:50]: Done

File: submissions/unabyband/popup-task/index.html (hunk 8 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html:26]: Done

File: submissions/unabyband/popup-task/main.css (hunk 16 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:80]: Anyway, it doesn't needed anymore :)
Done

File: submissions/unabyband/popup-task/main.css (hunk 12 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:5]: Thanks for advice, done!

File: submissions/unabyband/popup-task/main.css (hunk 17 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:79]: Yeah, I'm agree, thanks for your notice.

File: submissions/unabyband/popup-task/main.css (hunk 14 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:51]: Yeah, the idea is a bit weird, fixed!

File: submissions/unabyband/popup-task/main.css (hunk 10 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:172]: Fixed!

File: submissions/unabyband/popup-task/index.html (hunk 6 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html:10]: Ok, done! Thanks for your advice.

File: submissions/unabyband/popup-task/index.html (hunk 7 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/index.html:12]: Fixed

File: submissions/unabyband/popup-task/main.css (hunk 9 above)
Comment (unabyband) [reply submissions/unabyband/popup-task/main.css:149]: You're right, `link` is much better!

**Comment by unabyband** [comment]
I've tried to do my best, hope nothing forgot.
Re-view please one more time.

File: submissions/unabyband/popup-task/index.html (hunk 6 above)
Comment (kasionio) [reply submissions/unabyband/popup-task/index.html]: not done fully :) we have a header and a nav inside

File: submissions/unabyband/popup-task/main.css [hunk 19]
```diff
@@ -0,0 +1,293 @@
...
+.top_menu {
+    display: flex;
+    align-items: center;
+    padding: 0 15px;
+    list-style: none;
+}
+
+.top_menu__item {
+    line-height: 30px;
+    padding: 0 10px 0 10px;
```
Comment (kasionio) [inline submissions/unabyband/popup-task/main.css:29]: the same, short form - `padding: 0 10px`

**Review by kasionio — APPROVED** [review]
Good job! Let's move forward 🚀

**Pull request merged**